opsdroid/helper.py

```diff
@@ -44,6 +44,67 @@ def del_rw(action, name, exc):
     os.remove(name)
 
 
+# This is meant to provide backwards compatibility for versions
+# prior to  0.16.0 in the future this will be deleted
+
+
+def convert_dictionary(modules):
+    """Convert dictionary to new format.
+
+    We iterate over all the modules in the list and change the dictionary
+    to be in the format 'name_of_module: { config_params}'
+
+    Args:
+        modules (list): List of dictionaries that contain the module configuration
+
+    Return:
+        List: New modified list following the new format.
+
+    """
+    config = dict()
+
+    try:
+        for module in modules:
```

Comment:
If the user is using the new format and modules is already a dict then this will have some strange behaviour.

We should check if modules is already a dict and just return it if it is.

We could also add a deprecation warning or some other warning notice if it is a list and point people to the documentation on how to migrate.